risc0/zkp/src/prove/soundness.rs

```diff
@@ -0,0 +1,292 @@
+        let h_plus = self.trace_domain_size + self.biggest_combo;
+
+        let numerator = self.d * (h_plus - 1.0) + (self.trace_domain_size - 1.0);
+        let denominator = self.ext_field_size - self.trace_domain_size - self.lde_domain_size;
+        l_plus * numerator / denominator
+    }
+
+    fn e_deep_ali(&self, l_plus: f32) -> f32 {
+        self.e_deep(l_plus) + self.e_ali(l_plus)
+    }
+}
+
+#[cfg(test)]
+mod toy_model {
+    use super::*;
+    fn toy_model(coeffs_size: usize) {
+        // e, field extension degree
+        let ext_size = baby_bear::ExtElem::EXT_SIZE;
+        let field_size = baby_bear::P as f32;
+        let ext_field_size = libm::powf(field_size, ext_size as f32);
+        let trace_domain_size = (coeffs_size / ext_size) as f32;
+        let lde_domain_size = trace_domain_size * INV_RATE as f32;
+
+        let num_folding_rounds = num_folding_rounds(coeffs_size, ext_size);
+
+        let params = Params {
+            n_sigma_mem: 0,
+            n_sigma_bytes: 0,
+            n_trace_polys: 1.0,
+            d: 1.0,
+            biggest_combo: 1.0,
+            ext_size: 4,
+            ext_field_size,
+            trace_domain_size,
+            lde_domain_size,
+            num_folding_rounds,
+        };
+
+        pub fn toy_proven(params: Params) -> f32 {
+            let e_proximity_gap = params.e_proximity_gap_proven();
+
+            // α = (1 + 1/2m) * sqrt(ρ)
+            let alpha = (1.0 + 1.0 / (2.0 * M)) * RHO.sqrt();
+
+            let theta = 1.0 - alpha;
+            let l_plus = {
+                let rho_plus =
+                    (params.trace_domain_size + params.biggest_combo) / params.lde_domain_size;
+                let m_plus = 1.0 / (params.biggest_combo * (alpha / rho_plus.sqrt() - 1.0));
+                let m_plus = m_plus.ceil();
+
+                (m_plus + 0.5) / rho_plus.sqrt()
+            };
+            toy_soundness_error(params, theta, e_proximity_gap, l_plus)
+        }
+
+        pub fn toy_conjectured(params: Params) -> f32 {
+            let theta = 1.0 - RHO - ETA;
+            let e_proximity_gap = params.e_proximity_gap_conjectured();
+            let l_plus = {
+                let rho_plus =
+                    (params.trace_domain_size + params.biggest_combo) / params.lde_domain_size;
+                let epsilon_plus = 1.0 - rho_plus - theta;
+                let c_rho = 1; // unspecified exponent parameter in DEEP-FRI, Conjecture 2.3
+                (params.lde_domain_size / epsilon_plus).powi(c_rho)
+            };
+            toy_soundness_error(params, theta, e_proximity_gap, l_plus)
+        }
+
+        fn toy_soundness_error(
+            params: Params,
+            theta: f32,
+            e_proximity_gap: f32,
+            l_plus: f32,
+        ) -> f32 {
+            let plonk_plookup_error = params.plonk_plookup_error();
+            let fri_error = params.e_fri(theta, e_proximity_gap);
+            let deep_ali_error = params.e_deep_ali(l_plus);
```

Comment:
It looks to me like the analysis here is more complicated than what is done in ethSTARK and miden's toy problem analysis. 

in particular, I don't believe the other projects include `deep_ali_error` in the analysis for the toy problem conjecture.

Reply:
from ethstark: <img width="853" alt="image" src="https://github.com/risc0/risc0/assets/15272444/c16a7555-1628-4f24-bbfb-2a9d73a7935d">

Reply:
from miden: 
<img width="707" alt="image" src="https://github.com/risc0/risc0/assets/15272444/908b44bb-8ae5-4755-9cf0-180b1d91898a">

https://github.com/Al-Kindi-0/winterfell/blob/main/air/src/proof/mod.rs